the-collab-lab/tcl-66-smart-shopping-list
# 16. Layout Changes and Tailwind Implementation

## Description

- Installed Tailwind and react-icons for creating a cleaner ui
- Created a `tailwind.config` file with some updated variables for screen sizing and fonts
- Implemented figma layout design and styles for main content and basic sidebar
- Removed `manage-list` view from nav bar, and moved `addItem` logic from that view to the list view
    - Did keep the `manage-list` file because it had the invite feature, and we might want to migrate that somewhere else, but currently there is no share button displayed anywhere
- Added a welcome message with user display name to `home` view
- Updated styling to be responsive based on screen size
    - Note: I did not implement a nav bar at the top when the screen was small because creating the logic for having the nav bar become a menu button seemed outside of the scope of this. That seemed like it would be it's own ticket after delving deeper.
- Updated `helpers` to have hex codes for colours to be more accessible in contrast
- Added `capitalizeFirstLetterOfEachWord` function to capitalize first letter of each word for list items when rendering list
- Added hover effect for nav bar items and items in list
- Updated strikethrough to be a different background colour and "checked off" badge for better readability

Note: I did not style buttons and only styled inputs in order to be more responsive.

It's a lot of code changes, sorryyyyyyy <3

## Related Issue

closes #38 

## Acceptance criteria
- [x] The layout resembles the figma design and basic theming
- [x] The layout should be responsive based on screen sizes
- [x] TailwindCSS should be implemented for overall app consistency

## Type of Changes

accessibility, enhancement, and refactoring

## Updates

### Before

<img width="1118" alt="Screenshot 2024-03-24 at 6 42 21 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/4f173a0c-fffb-42fc-951c-44f3854d23ec">
<img width="1116" alt="Screenshot 2024-03-24 at 6 42 36 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/7dcd7850-e5b8-446f-948a-3cd33b589bb8">

### After

<img width="1118" alt="Screenshot 2024-03-24 at 6 43 15 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/acc658f3-80ad-495c-abf1-fdfa4cc1b2f6">
<img width="1119" alt="Screenshot 2024-03-24 at 6 43 01 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/b2bda26d-4f0d-4cc5-9663-8d1099ff3320">

Updated UI of list with no items:
<img width="1118" alt="Screenshot 2024-03-24 at 6 43 31 PM" src="https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/98231995/a44330cd-15cc-42b7-ac35-5dedcd050a2c">

## Testing Steps / QA Criteria

1. git pull from main
2. git checkout an-layout
3. npm install
4. navigate to localhost
5. click links, add items, delete items, do whatever
6. create a new list or go to a list without items to see new UI for adding first item
7. let me know if I am missing something :)
